org-status.py

```diff
@@ -1,74 +1,86 @@
-from IGitt.GitHub.GitHubOrganization import GitHubOrganization
 
-error = 'error'
-unknown = 'unknown'
-failing = 'failing'
-passing = 'passing'
+from status_providers import Status
+from org_hosts import get_all_supported_hosts
 
-def process_travis_result(html):
-  html = html.lower()
 
-  for status in [passing, failing, unknown, error]:
-    if status in html:
-      return status    
+def get_host_token(host_name):
+    return environ['{}_TOKEN'.format(host_name.upper())]
 
-def get_travis_status(repo, branch='master'):
-  url = 'https://api.travis-ci.org/coala/{repo}.svg?branch={branch}'.format(
-          repo=repo, branch=branch)
 
-  svg_result = requests.get(url)
-  svg = svg_result.text
+def get_org_host(host_org):
+    parts = host_org.strip().split(':')
 
-  return process_travis_result(svg)
+    # defaults to gh
+    if len(parts) == 1:
+        parts = ['github', parts[0]]
 
-def process_repo(repo):
-  repo_name = repo.web_url.split('/')[-1]
-  return (repo.web_url, get_travis_status(repo_name))
+    host, org = parts[:2]
 
-def get_org_repo_status(token, org_name):
-  gh_org = GitHubOrganization(token, org_name)
+    for supported_host in get_all_supported_hosts():
+        if host == supported_host.HostName:
+            return (supported_host, org)
+
+
+def aggregate_org_status(org_host, threads=2):
+    with Pool(processes=threads) as pool:
+        return pool.map(org_host.process_repository, org_host.repositories)
+
+
+def present_status(statuses, no_color):
+    color = (lambda l, *_: l) if no_color else colored
+    r_pass, r_fail, r_unknown, r_error = 0, 0, 0, 0
+
+    for status in statuses:
+        repo_status = status.repo_status or Status.UNDETERMINED
+        status_text = repo_status.value
+
+        if repo_status == Status.PASSING:
+            status_text = color(status_text, 'green')
+            r_pass += 1
+        elif repo_status == Status.FAILING:
+            status_text = color(status_text, 'red')
+            r_fail += 1
+        elif repo_status == Status.ERROR:
+            status_text = color(status_text, 'magenta')
+            r_error += 1
+        elif repo_status in [Status.UNKNOWN, Status.UNDETERMINED]:
+            r_unknown += 1
+            continue
+
+        print('{repo}: {status}'.format(
+            repo=status.repo_url, status=status_text))
+
+    print('{} Passing, {} Failing, {} Error, {} Unknown '
+          'of {} Repositories'.format(
+            r_pass, r_fail, r_error, r_unknown, len(statuses)))
+
+
+def get_argument_parser():
+    parser = ArgumentParser()
+    parser.add_argument('org', help='host:org')
+    parser.add_argument('--threads', type=int, default=2)
+    parser.add_argument('--no-color', action='store_true')
+    parser.add_argument('--verbose', '-v', action='store_true')
+
+    return parser
+
+
+def main():
+    parser = get_argument_parser()
+    args = parser.parse_args()
+
+    Host, org = get_org_host(args.org)
+    token = get_host_token(Host.HostName)
```

Comment:
would be good to catch `KeyError: 'GITHUB_TOKEN'`

and give slightly user-friendly error